pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotResultSet.java

```diff
@@ -0,0 +1,1227 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.client;
+
+import java.io.InputStream;
+import java.io.Reader;
+import java.math.BigDecimal;
+import java.net.URL;
+import java.sql.Array;
+import java.sql.Blob;
+import java.sql.Clob;
+import java.sql.Date;
+import java.sql.NClob;
+import java.sql.Ref;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.RowId;
+import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
+import java.sql.SQLWarning;
+import java.sql.SQLXML;
+import java.sql.Statement;
+import java.sql.Time;
+import java.sql.Timestamp;
+import java.util.Calendar;
+import java.util.HashMap;
+import java.util.Map;
+
+
+public class PinotResultSet implements ResultSet {
+  private final org.apache.pinot.client.ResultSet _resultSet;
+  private int _totalRows;
+  private int _currentRow;
+  private Map<String, Integer> _columns = new HashMap<>();
+
+  public PinotResultSet(org.apache.pinot.client.ResultSet resultSet) {
```

Comment:
Any reason why this has to use pinot.client.ResultSet (which seems like a really old class). We have ResultTable used for returning responses that is a more recent code, you may want to check it out.

Reply:
+1, lets use resultTable. Its ok to say we support jdbc only on SQL endpoint

Reply:
Since I am using Java client to run queries, it is currently returning ResultSet.
Should I create a seperate PR making changes in the Java client first so that it returns ResultTable?

Reply:
sorry, my bad, ResultSet is still the right interface for the client. We need to deprecate few methods like 

```
  int getGroupKeyLength();
  String getGroupKeyColumnName(int groupKeyColumnIndex);
  int getGroupKeyInt(int rowIndex, int groupKeyColumnIndex);
  long getGroupKeyLong(int rowIndex, int groupKeyColumnIndex);
  float getGroupKeyFloat(int rowIndex, int groupKeyColumnIndex);
  double getGroupKeyDouble(int rowIndex, int groupKeyColumnIndex);
  String getGroupKeyString(int rowIndex, int groupKeyColumnIndex);
```

jdbc driver code is not using any of them. so we are good here.